### ISSUE TITLE
Editor completion stops working after changing interpreter

### ISSUE THREAD

**Issue by trianta2** [spyder-ide/spyder#16812]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [x] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [x] Reset preferences with ``spyder --reset``
    * [x] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

I've recently installed Spyder in a separate new conda environment due to project dependency issues. I pointed Spyder to use the interpreter in my project's conda env, and then tab completion stopped working within the editor.

For example, `tornado` is installed in both the spyder and project environments. If the default Spyder interpreter is used, then I can tab complete the import of `tornado` as well as its contents. If I switch the interpreter to my project env, `tornado` no longer shows up as a candidate for import tab completion, nor do its contents.

However, the console doesn't appear to be affected; completion in the console still works regardless of the interpreter. 

FWIW, I used `pyenv` to bootstrap my system with Python environments, including anaconda. Not sure if nested managers could be causing the problem.

### What steps reproduce the problem?

1. Use default (Spyder) interpreter
2. In the editor, try using tab completion to import an installed library, e.g. `tornado`
3. Change interpreter to another conda env
4. Repeat (2)

### What is the expected output? What do you see instead?

The `tornado` library should show up in step (4), but it doesn't.

## Versions

* Spyder version: 5.1.5 None
* Python version: 3.7.12 64-bit
* Qt version: 5.12.9
* PyQt5 version: 5.12.3
* Operating System: Darwin 18.6.0

### Dependencies

```

# Mandatory:
applaunchservices >=0.1.7     :  0.2.1 (OK)
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.0.2 (OK)
IPython >=7.6.0               :  7.29.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.0 (OK)
jsonschema >=3.2.0            :  4.2.1 (OK)
keyring >=17.0.0              :  23.2.1 (OK)
nbconvert >=4.0               :  6.2.0 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
parso >=0.7.0;<0.9.0          :  0.8.2 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.10.0 (OK)
pylint >=2.5.0;<2.10.0        :  2.7.2 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.2.2;<1.3.0          :  1.2.4 (OK)
pylsp_black >=1.0.0           :  None (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.1 (OK)
qtawesome >=1.0.2             :  1.1.0 (OK)
qtconsole >=5.1.0             :  5.1.1 (OK)
qtpy >=1.5.0                  :  1.11.2 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  58.5.3 (OK)
sphinx >=0.6.6                :  4.3.0 (OK)
spyder_kernels >=2.1.1;<2.2.0 :  2.1.3 (OK)
textdistance >=4.2.0          :  4.2.2 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
zmq >=17                      :  22.3.0 (OK)

# Optional:
cython >=0.21                 :  None (OK)
matplotlib >=2.0.0            :  None (OK)
numpy >=1.7                   :  None (OK)
pandas >=1.1.1                :  None (OK)
scipy >=0.17.0                :  None (OK)
sympy >=0.7.3                 :  None (OK)
```

**Comment by steff456** [comment]
Hi @trianta2,

Just to be clear, you created this new environment using `pyenv` or `conda`? I think it is highly probable that Spyder is confusing itself if you are nesting the managers. Can you try to just use `conda` to create a new environment and link that one to the default Spyder?

**Comment by trianta2** [comment]
Hi @steff456, what I did was:

1) Use `pyenv` to install miniconda3-4.7.12
2) Use `pyenv` to set miniconda3-4.7.12 as my global Python environment
3) Use the now global `conda` manager to create new `conda` environments, e.g. `spyder5` and `project`.

The reason for the nested managers is that sometimes I need multiple versions of Python available locally, e.g. when I need `tox` to run against multiple Python versions. `conda` does not offer me this flexibility, but `pyenv` does via the `local` command.

I disabled `pyenv` and only had `conda` available. The import error I described above still happens. I have some additional findings:

Debugging environments created via `conda`:
* `spyder5`: python=3.7.11, spyder=5.1.5, spyder-kernels=2.1.3
* `project`: python=3.7.11, spyder-kernels=2.1.3, numpy
* `project-py36`: python=3.6.13, spyder-kernels=2.1.1

Findings:
* If the Python versions are identical, e.g. as in environments `spyder5` and `project`, then import tab completion works for libraries installed in **both** environments, e.g. `tornado`.
* However, if the Python interpreter is changed to `project`, and `project` has a library that `spyder5` does not have, e.g. `numpy`, then import tab completion of `numpy` **does not work**.
* If the Python versions are not identical, e.g. as in environments `spyder5` and `project-py36`, then import tab completion of a shared library, e.g. `tornado`, **does not work**.

In other words:

| Python interpreter  | Import tab completion works for `tornado`? | Import tab completion works for `numpy`? |
| ------------- | ------------- |------------- |
| `spyder5` (default)  | ✅  | NA | 
| `project`  | ✅  | ❌ | 
| `project-py36`  | ❌  | ❌ |

**Comment by steff456** [comment]
Hi @trianta2,

Sadly we are unable to support this type of installation because the completions will track the python installed on the environment, and in this case as you have nested environments I don't know which one is grabbing. I understand you have certain gains from using `pyenv` and `conda` together, but it is a complicated mix because both of them want to register a global Python. This is why we are unable to help you with this.

**Comment by trianta2** [comment]
For the table I posted above, I had disabled `pyenv` and only had `conda` initialized, and I still had import issues. I can install a miniconda distribution in isolation, without pyenv, and see if I still have any issues.
